### DISCUSSION TITLE
Do you know what's the format of the stderr/stdout of leelaz?

### DISCUSSION THREAD

**Discussion by HackYardo** [kaorahi/lizgoban#79] [Q&A] [answered]
I'm writing a GTP Web GUI named "VastGo" for fun, I want to run not only KataGo but also GNUGo and Leela-Zero as well.
The output of KataGo and GNUGo are **plain string in terminal and Sublime editor**, but the Leela-Zero's are  **plain string in terminal and *mystical format?* in Sublime editor**, and there are **NO** document or issue about in Leela-Zero's repo. See below (the command is `leelaz -g -w path/to/weight.gz -v 361 --noponder --cpu-only`):
In terminal (without stderr):
```bash
quit
=

```
In Sublime:
```
0000 0000 0000 0000 0000 0000 0000 0000
0000 0000 0000 0000 0000 0000 0000 0000
0000 0000 0000 0000 0000 0000 0000 0000
0000 0000 0000 0000 0000 0000 0000 0000
0000 0000 0000 0000 0000 0000 0000 0000
0000 0000 0000 0000 0000 0044 6574 6563
7469 6e67 2072 6573 6964 7561 6c20 6c61
7965 7273 2e2e 2e76 312e 2e2e 3132 3820
6368 616e 6e65 6c73 2e2e 2e36 2062 6c6f
636b 732e 0a49 6e69 7469 616c 697a 696e
6720 4350 552d 6f6e 6c79 2065 7661 6c75
6174 696f 6e2e 0a53 6574 7469 6e67 206d
6178 2074 7265 6520 7369 7a65 2074 6f20
3433 3630 204d 6942 2061 6e64 2063 6163
6865 2073 697a 6520 746f 2034 3834 204d
6942 2e0a 7175 6974 0a3d 200a 0a
```
In `cat` command:
```bash
Detecting residual layers...v1...128 channels...6 blocks.
Initializing CPU-only evaluation.
Setting max tree size to 4360 MiB and cache size to 484 MiB.
quit
=

```
**The three above are all wrong!!!** It should be (with stderr):
```
Using 8 thread(s).
RNG seed: 14907218604829807376
BLAS Core: built-in Eigen 3.3.7 library.
Detecting residual layers...v1...128 channels...6 blocks.
Initializing CPU-only evaluation.
Setting max tree size to 4360 MiB and cache size to 484 MiB.
quit
=

```
What's the format of the stdout/stderr of Leela-Zero?

**Comment by kaorahi** [comment, marked as answer]
not sure what you are asking... You seem to have some trouble when you tried to use Leela Zero. But I cannot understand what is your trouble (what you did on what environment, why you think it is a trouble, ...). If you are asking about Sublime Text, I can answer nothing as I've never used it.

For the protocol, see [GTP](http://www.lysator.liu.se/~gunnar/gtp/) and "lz-analyze" in [GTP_Extensions.md](https://github.com/lightvector/KataGo/blob/v1.11.0/docs/GTP_Extensions.md). These documents (and some trials on a terminal) should be enough to know how to use Leela Zero. GTP consoles on Lizzie or Sabaki may be also helpful though they hide the outputs of "lz-analyze".

**Reply by HackYardo** [reply]
- Thank you tons! It is a bit bad to learn LZ via KG, LZ should have a doc. 
- Yeah, I want to implement a GTP console like Sabaki or Lizzie by putting std IOE streams in a file. 
- The story of the mystery format: I write the LZ's stderr to a file in binary format and the stdout in plain text format, so that the file is not text or binary format, it is a mystery format for sure. 
- Thank you a lot.
